scripts/sync-up-helm-chart.sh

```diff
@@ -32,7 +32,37 @@ if [ -n "${crd_diff}" ]; then
   cp -f $INSTALL_CRDS_YAML $CHART_FEDERATED_CRD_DIR/crds.yaml
 fi
 
+# Start kube-apiserver to generate consistent CRD validations
+./bin/etcd > /dev/null 2>&1 &
+sleep 5
+./bin/kube-apiserver --etcd-servers=http://127.0.0.1:2379 --service-cluster-ip-range=10.0.0.0/16 --cert-dir ${CHART_FEDERATED_CRD_DIR} > /dev/null 2>&1 &
+# Wait until kube-apiserver is ready
+sleep 10
+
+# Generate fake kubeconfig to access kube-apiserver. It is cleaned when script is done.
+cat <<EOF > ${CHART_FEDERATED_CRD_DIR}/kubeconfig
+apiVersion: v1
+clusters:
+- cluster:
+    server: 127.0.0.1:8080
+  name: development
+contexts:
+- context:
+    cluster: development
+    user: ""
+  name: federation
+current-context: ""
+kind: Config
+preferences: {}
+users: []
+EOF
 # Generate YAML templates to enable resource propagation for helm chart.
 for filename in ./config/federatedirectives/*.yaml; do
-  ./bin/kubefed2 federate enable -f "${filename}" --federation-namespace="${NS}" --dry-run -oyaml > ${CHART_FEDERATED_PROPAGATION_DIR}/$(basename $filename)
+  ./bin/kubefed2 --kubeconfig ${CHART_FEDERATED_CRD_DIR}/kubeconfig federate enable -f "${filename}" --federation-namespace="${NS}" --host-cluster-context federation --dry-run -o yaml > ${CHART_FEDERATED_PROPAGATION_DIR}/$(basename $filename)
 done
+
+# Clean kube-apiserver daemons and temporary files
+kill %1 # etcd
+kill %2 # kube-apiserver
+rm -f $CHART_FEDERATED_CRD_DIR/kubeconfig $CHART_FEDERATED_CRD_DIR/apiserver.crt $CHART_FEDERATED_CRD_DIR/apiserver.key
+
```

Comment:
kill this

Reply:
I'd like to keep this because pre-commit.sh may be run locally to make sure code change is ready. In such case, clean the generated information is useful.

Reply:
blank line deleted